service/controller/v13/resource/instance/create_deployment_uninitialized.go

```diff
@@ -0,0 +1,91 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/azure-operator/service/controller/v13/blobclient"
+	"github.com/giantswarm/azure-operator/service/controller/v13/controllercontext"
+	"github.com/giantswarm/azure-operator/service/controller/v13/key"
+	"github.com/giantswarm/azure-operator/service/controller/v13/resource/instance/internal/state"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+)
+
+func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+	deploymentsClient, err := r.getDeploymentsClient(ctx)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring deployment")
+
+	computedDeployment, err := r.newDeployment(ctx, customObject, nil)
+	if controllercontext.IsInvalidContext(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "missing dispatched output values in controller context")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "did not ensure deployment")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return currentState, nil
+	} else if blobclient.IsBlobNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ignition blob not found")
+		resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
Sorry for the late review but I was busy with other stuff. I wanted to get to this eventually. And I like it. There is one but though. This isn't inline with this doc anymore: https://github.com/giantswarm/giantswarm/blob/895f0abcf3a97835c76d71d8967ef7c3a22a7e7b/sigs/operator/resource_control_flow.md. I'm fine with changing the doc accordingly.

Reply:
Ack. I'll think about that and provide a PR for it. So far this approach is only used inside this single resource, but makes sense to document it in case it's useful somewhere else as well.